net/dns/manager_darwin.go

```diff
@@ -8,12 +8,16 @@ import (
 	"os"
 
 	"go4.org/mem"
+	"tailscale.com/control/controlknobs"
 	"tailscale.com/health"
 	"tailscale.com/types/logger"
 	"tailscale.com/util/mak"
 )
 
-func NewOSConfigurator(logf logger.Logf, health *health.Tracker, ifName string) (OSConfigurator, error) {
+// NewOSConfigurator creates a new OS configurator.
+//
+// health and knobs may be nil.
```

Comment:
[Nit](https://go.dev/wiki/CodeReviewComments#comment-sentences): The health tracker may be nil; the knobs are ignored on this platform.

(or words to that effect—applies elsewhere as relevant too)

Reply:
Thanks for the suggestion! I went with your wording but also added that the knobs may be nil. I believe this bit might be useful if we start using knobs in `NewOSConfigurator` on the platforms where it is currently unused.

Reply:
SGTM. I agree it's worth mentioning that they can be nil for the cases where the parameter is usable.